tron/kubernetes.py

```diff
@@ -338,22 +349,25 @@ def handle_next_event(self, _=None) -> None:
         * control: events regarding how the task_proc plugin is running - handled directly
         * task: events regarding how the actual tasks/Pods we're running our doing - forwarded to KubernetesTask
         """
-        if self.deferred and not self.deferred.called:
+        if self.deferred is not None and not self.deferred.called:
             log.warning("Already have handlers waiting for next event in queue, not adding more")
             return
 
         self.deferred = self.queue.get()
-
+        if self.deferred is None:
+            log.warning("Unable to get a handler for next event in queue - this should never happen!")
+            # TODO: figure out how to recover if we were unable to get a handler
+            # Not adding a callback is very bad here as this means we will never handle this event
```

Comment:
super-nit if we want to be extra verbose for future-us:
```suggestion
            # Not adding a callback is very bad here as this means we will never handle future events
            # This hasn't happened to date, and if it does hopefully the errback added in a previous
            # handle_next_event() will get us back on track for now 
```